ckanext/geodatagov/tests/test_db_solr_sync.py

```diff
@@ -91,24 +91,22 @@ def create_datasets(self):
 
         # Case 5 - changing harvest_object_id in DB makes Solr out of date
         # Solr starts with the same id as the current harvest_object.id
-        assert get_solr_hoid(self.dataset5['id']) == dataset5_hoid.id
+        assert get_solr_hoid(self.dataset5['id']) == ho5.id
 
         # mark the current harvest_object outdated
-        dataset5_hoid.current = False
-        dataset5_hoid.save()
+        ho5.current = False
+        ho5.save()
 
         # a new harvest_object with a new id.
-        new_dataset5_hoid = HarvestObject(
-            id='newid',
-            package_id=dataset5_hoid.package_id,
-            job=dataset5_hoid.job,
-            current=True
-        )
-        new_dataset5_hoid.save()
+        add_harvest_object(self.dataset5, "newid")
 
         # Solr is unaware of the new id
         assert get_solr_hoid(self.dataset5['id']) != 'newid'
```

Comment:
can you walk me through the sketch for test of `dataset5`?

Reply:
Sure. In the beginning dataset5 is just like other normal datasets, one dataset with one valid harvest object, DB and SOLR are in sync. But later on we mark its harvest object not current, and give it a new harvest object. At this point SOLR is not aware of this update. We test that the script can detect the difference between dataset5 in DB and dataset5 in Solr, and update SOLR with the new harvest object for dataset5.

Reply:
> and update SOLR with the new harvest object for dataset5.

this is the part I don't see happening, or i'm missing it...

Reply:
In the beginning all is good:
```
        # Solr starts with the same id as the current harvest_object.id
        assert get_solr_hoid(self.dataset5['id']) == ho5.id
```
Then we verify SOLR has bad info after we changed dataset5's ho:
```
        # Solr is unaware of the new id
        assert get_solr_hoid(self.dataset5['id']) != 'newid'
```
Then we run db-solr-sync script:
```
        raw_cli_output = runner.invoke(
            cli.db_solr_sync,
            args=[],
        )
```
In the end, we verifify SOLR has good info:

```
        assert get_solr_hoid(self.dataset5['id']) == "newid"
```